### PR TITLE
exporter plugin without evernote

### PR DESCRIPTION
**Pull request by pazos**
From the PoV of the commit history doing everything on a single step (rename evernote.koplugin -> exporter.koplugin and remove specific evernote functions) looks better.

That requires to skip the deprecate first, remove later step.

The other alternative, sketched here, is add a clone of the "evernote" plugin without evernote specific functions, mark the original plugin as deprecated and tell the users to use the new one.

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/7983)

### CODE REVIEW

**Comment by poire-z** [comment]
Looking at the `Files changed`, I was going to write that I apologize for not reviewing all of this because I have little interest in all this... :/
But just applying this PR and `diff -Naur plugins/evernote.koplugin plugins/exporter.koplugin`, it looks a lot less scary ! :)
It's mostly a bunch of renames like:
```diff
-function EvernoteExporter:migrateClippings()
+function Exporter:migrateClippings()
```
and some functions and menu items removed:
```
EvernoteExporter:login()
EvernoteExporter:doLogin(username, password)
EvernoteExporter:logout()
EvernoteExporter:exportBooknotesToEvernote(client, title, booknotes)
```

So, I still don't care much about all this, but it looks rather fine, and yes, your suggestion to git rename will make it really looks as what I discovered above :)

File: plugins/exporter.koplugin/main.lua [hunk 1]
```diff
@@ -0,0 +1,578 @@
...
+local N_ = _.ngettext
+local slt2 = require('slt2')
+local MyClipping = require("clip")
+local json = require("json")
+
+local Exporter = InputContainer:new{
+    name = "evernote",
+    login_title = _("Login to Evernote"),
+    notebook_name = _("KOReader Notes"),
+    evernote_domain = nil,
```
Comment (yparitcher) [inline plugins/exporter.koplugin/main.lua:23]: Evernote leftovers?

Comment (pazos) [reply plugins/exporter.koplugin/main.lua:23]: Yup. Thanks!

**Comment by pazos** [comment]
@poire-z: TBH I don't care much about all this too. I don't use the plugin at all. Just want to murder the evernote bits.

>  your suggestion to git rename will make it really looks as what I discovered above :)

Probably makes sense to git rename and murder the old one in a single step. The missing part is migrating the old bits of relevant data from the "evernote" table to a new "exporter" table. Mostly useful for joplin settings, I think, as the rest is trivial to setup from scratch.

**Comment by pazos** [comment]
> Looks pretty straightforward. Btw, "Export highlights" or "Highlight exporter"?

As you wish. Bad at naming I am :p

**Comment by mergen3107** [comment]
> "Export highlights"

I'd vote for this

File: plugins/exporter.koplugin/_meta.lua [hunk 2]
```diff
@@ -0,0 +1,6 @@
+local _ = require("gettext")
+return {
+    name = "exporter",
+    fullname = _("Export highlights"),
+    description = _("Exports highlights and notes"),
```
Comment (Frenzie) [inline plugins/exporter.koplugin/_meta.lua:5]:
```suggestion
    description = _("Exports highlights and notes."),
```

**Comment by poire-z** [comment]
Some issue with CircleCI since this was merged, in the docs-and-coverage part:
```
Checking out koreader/doc for update.
Cloning into 'koreader_doc'...
remote: Enumerating objects: 174, done.        
remote: Counting objects: 100% (136/136), done.        
remote: Compressing objects: 100% (15/15), done.        
remote: Total 174 (delta 121), reused 136 (delta 121), pack-reused 38        
Receiving objects: 100% (174/174), 156.84 KiB | 0 bytes/s, done.
Resolving deltas: 100% (142/142), done.
Checking connectivity... done.
~/project/doc ~/project
reading configuration from config.ld
Failed to generate documents...

Exited with code exit status 1

CircleCI received exit code 1
```

**Comment by hius07** [comment]
As I noticed, the exporter config file is opened every time when a book is opened (in init).
https://github.com/koreader/koreader/blob/4fd4861d090fdc156e302efed8e53a7fcfbe58ae/plugins/exporter.koplugin/main.lua#L76

Can we avoid this and open config when it is needed only (two places)?
https://github.com/koreader/koreader/blob/4fd4861d090fdc156e302efed8e53a7fcfbe58ae/plugins/exporter.koplugin/main.lua#L305-L307
https://github.com/koreader/koreader/blob/4fd4861d090fdc156e302efed8e53a7fcfbe58ae/plugins/exporter.koplugin/main.lua#L392-L397

**Comment by poire-z** [comment]
> Can we avoid this and open config when it is needed only (two places)?

Feels ok.

**Pull request merged**